turbo/stages/headerdownload/header_algos.go

```diff
@@ -608,6 +608,9 @@ func (hd *HeaderDownload) InsertHeader(hf FeedHeaderFunc, terminalTotalDifficult
 			blocksToTTD = x.Uint64()
 		}
 	}
+	if hd.stageSyncStep > 0 && hd.highestInDb%hd.stageSyncStep == 0 {
```

Comment:
why put it here, could be the same place as stageSyncUpperBound

Reply:
If put as the same place above, then it was check before insert, then the blockHeight whose mod stageSyncStep ==0 will never be inserted into db, so put here, insert before check.